### PR TITLE
fix #578 Switch "fixed" connection pool by default for TCP/HTTP client

### LINKED ISSUE #578: Default back connection pool to a fixed limit (e.g. 1000)
Currently Connection Pooling uses elastic (unbounded) default value. Local VM might struggle with this default, we should investigate whether this is a correct default to assume or if we limits should be given and tuning documented.

### PR DESCRIPTION
**Pull request by violetagg**

### CODE REVIEW

**Comment by violetagg** [comment]
@Buzzardo Can you take a look at the documentation? Thanks

@smaldini Do we want idle time by default?

The max number of the channels in the pool will be 500 with 45s acquisition timeout

File: src/docs/asciidoc/tcp-client.adoc [hunk 1]
```diff
@@ -358,6 +362,37 @@ public class Application {
...
+                         .connectNow();
+
+        connection.onDispose()
+                  .block();
+    }
+}
+----
+====
+
+NOTE: Be cautious with "`elastic`" connection pool when you expect a high load. You might experience
```
Comment (Buzzardo) [inline src/docs/asciidoc/tcp-client.adoc]: Add "an" after "with" on this line, please.

File: src/docs/asciidoc/tcp-client.adoc [hunk 2]
```diff
@@ -302,11 +302,15 @@ public class Application {
 
 == Connection Pool
 
-By default, the `TCP` client uses an "`elastic`" connection pool. This means that the implementation creates a new channel
-if someone tries to acquire a channel but none is in the pool. Also, there is no limit on the maximum concurrent channels.
+By default, the `TCP` client uses a "`fixed`" connection pool with `500` as a maximum number of the channels and
```
Comment (Buzzardo) [inline src/docs/asciidoc/tcp-client.adoc]: "a maximum number" should be "the maximum number".

File: src/docs/asciidoc/tcp-client.adoc [hunk 3]
```diff
@@ -302,11 +302,15 @@ public class Application {
...
+`45s` as acquisition timeout.
```
Comment (Buzzardo) [inline src/docs/asciidoc/tcp-client.adoc]: Please insert "the" before "acquisition" on this line.

File: src/docs/asciidoc/tcp-client.adoc [hunk 4]
```diff
@@ -302,11 +302,15 @@ public class Application {
...
+This means that the implementation creates a new channel if someone tries to acquire a channel but none is in the pool.
+When the maximum number of the channels in the pool is reached, new tries to acquire a channel will be delayed
```
Comment (Buzzardo) [inline src/docs/asciidoc/tcp-client.adoc]: Please change "will be" to "are".

File: src/docs/asciidoc/tcp-client.adoc [hunk 5]
```diff
@@ -302,11 +302,15 @@ public class Application {
...
+until a channel is returned to the pool again.
 The implementation uses `FIFO` order for channels in the pool regardless of whether it is "`elastic`" or "`fixed`".
+By default there is no idle time specified for the channels in the pool
```
Comment (Buzzardo) [inline src/docs/asciidoc/tcp-client.adoc]: Please add a comma after "default" and a period at the end.

File: src/docs/asciidoc/tcp-client.adoc [hunk 6]
```diff
@@ -302,11 +302,15 @@ public class Application {
...
 
-If you need a "`fixed`" connection pool, you can apply the following configuration by using
+If you need a "`elastic`" connection pool, you can apply the following configuration by using
```
Comment (Buzzardo) [inline src/docs/asciidoc/tcp-client.adoc]: Please change "need a" to "need an".

File: src/docs/asciidoc/tcp-client.adoc (hunk 1 above)
Comment (violetagg) [reply src/docs/asciidoc/tcp-client.adoc]: done

File: src/docs/asciidoc/tcp-client.adoc (hunk 2 above)
Comment (violetagg) [reply src/docs/asciidoc/tcp-client.adoc]: done

File: src/docs/asciidoc/tcp-client.adoc (hunk 3 above)
Comment (violetagg) [reply src/docs/asciidoc/tcp-client.adoc]: done

File: src/docs/asciidoc/tcp-client.adoc (hunk 4 above)
Comment (violetagg) [reply src/docs/asciidoc/tcp-client.adoc]: done

File: src/docs/asciidoc/tcp-client.adoc (hunk 5 above)
Comment (violetagg) [reply src/docs/asciidoc/tcp-client.adoc]: done

File: src/docs/asciidoc/tcp-client.adoc (hunk 6 above)
Comment (violetagg) [reply src/docs/asciidoc/tcp-client.adoc]: done

**Pull request merged**